Lab/src/main/java/agh/ics/oop/gui/App.java

```diff
@@ -1,53 +1,95 @@
 package agh.ics.oop.gui;
-import agh.ics.oop.Animal;
-import agh.ics.oop.GrassField;
-import agh.ics.oop.Vector2d;
+import agh.ics.oop.*;
 import javafx.application.Application;
+import javafx.application.Platform;
 import javafx.scene.Scene;
+import javafx.scene.control.Button;
 import javafx.scene.control.Label;
-import javafx.scene.layout.ColumnConstraints;
-import javafx.scene.layout.GridPane;
-import javafx.scene.layout.RowConstraints;
+import javafx.scene.control.TextField;
+import javafx.scene.layout.*;
 import javafx.stage.Stage;
-public class App extends Application{
+import org.w3c.dom.Text;
 
-//niestety nie zdążyłem dokończyć zadań
+import java.io.FileNotFoundException;
 
+public class App extends Application implements IMapUpdateObserver {
+    private GrassField map = new GrassField(10);
+    private GridPane grid = new GridPane();
+    private final int BOX_SIZE = 19;
+    private SimulationEngine engine;
+    public void init(){
+        Vector2d[] positions = {new Vector2d(3,4)};
+        engine = new SimulationEngine(new MoveDirection[0],map,positions);
+        engine.addObserver(this);
+        Thread thread = new Thread(engine);
```

Comment:
to chyba niepotrzebne, potem tworzymy wątek za każdym razem gdy ktoś kliknie start